### PR TITLE
Fix decrypt errors

### PR DESCRIPTION
**Pull request by Isha988**
I am working on solving decrypt errors, because it was creating problem in login, showing user image and was generating useless multiple decrypt errors in console.

Still decrypt errors exist in multiple files like profile component of dashboard.
Please tell me two things:
1. How to access mentor token in profile component ?
2. Mention other files in which decrypt error exist.

### CODE REVIEW

**Comment by anmode** [comment]
So when you logged in as mentor you will se a token is created into cookie named mentor_token
and when you hit protected endpoint:
like as update profile and any other endpoint which requires a authentication as get profile also comes in:
this token would be used, without this token those endpoint can't be accessed,

just apply middleware on those routes, i have created a middleware already

File: components/layout/UserProfile.js [hunk 1]
```diff
@@ -47,7 +46,7 @@ function UserProfile() {
   }
 
   //Decrypt Data:
-  const decryptedData = decryptData(
+  const decryptedData = JSON.parse(
     localStorage.getItem("userData") || localStorage.getItem("mentorData"),
   );
```
Comment (anmode) [inline components/layout/UserProfile.js:51]: remove... on login I saved all data in localstorage.. just access that.

File: components/layout/Header.js [hunk 2]
```diff
@@ -18,8 +17,8 @@ function Header() {
     isUserLoggedIn,
     setIsUserLoggedIn,
   } = useAuth();
-  const userData = decryptData(localStorage.getItem("userData"));
-  const mentorData = decryptData(localStorage.getItem("mentorData"));
+  const userData = JSON.parse(localStorage.getItem("userData"));
+  const mentorData = JSON.parse(localStorage.getItem("mentorData"));
```
Comment (anmode) [inline components/layout/Header.js:21]: On login parsed data already present se.... try to login please local backend

**Review by anmode — CHANGES_REQUESTED** [review]
Please use local backend not demo one!

**Comment by Isha988** [comment]
> just apply middleware on those routes, i have created a middleware already

Okay, have you also made some changes in database models?

**Comment by Isha988** [comment]
> Please use local backend not demo one!

Yes, I am using local backend in develop branch.

File: components/layout/UserProfile.js (hunk 1 above)
Comment (Isha988) [reply components/layout/UserProfile.js:51]: I am getting what are you saying, local storage is having json stringify object
```
 localStorage.setItem("mentorData", JSON.stringify(res.mentorData));
```
so there is a need to parse it before using.

Comment (anmode) [reply components/layout/UserProfile.js:51]: yes yes! you are correct

**Comment by anmode** [comment]
@Isha988 is this completed? rest look good to me

**Comment by Isha988** [comment]
> @Isha988 is this completed? rest look good to me

dashboard profile component is remaining, and maybe there are some more page those haven't countered till now.

**Comment by anmode** [comment]
> > @Isha988 is this completed? rest look good to me
> 
> dashboard profile component is remaining, and maybe there are some more page those haven't countered till now.

no worries woh dashboard m dekhte h abhi toh backend configure ni kia frontend se!

**Pull request merged**